### ISSUE TITLE
Address runtime warnings on xeus-cpp-lite while building with assertions

### ISSUE THREAD

**Issue by anutosh491** [compiler-research/xeus-cpp#250]
If xeus-cpp-lite is built with debug flags or a combination of them like (-O0 -g -s ASSERTIONS=2 -fsanitize=address)

![Image](https://github.com/user-attachments/assets/ba34fc78-73c9-49d7-b13a-89cafeaf8d2c)

There are warnings for duplicate symbols or something. 

Look into this and try getting rid of it !

**Comment by kr-2003** [comment]
Does passing ```-Wl,--allow-multiple-definition``` while building be a good approach to go with?

**Comment by anutosh491** [comment]
Where would you be adding it ?

Here are some questions 
1) What is the pool of symbols that are being duplicated ? And from where (as in what is source_1 and source_2 and other sources from where these symbols are being duplicated)

**Comment by kr-2003** [comment]
How can I reproduce these warnings? I have tried building ```xeus-cpp-lite``` using 
```
emcmake cmake   \  
      -DCMAKE_BUILD_TYPE=Debug             \                   
      -DCMAKE_INSTALL_PREFIX=$PREFIX           \                 
      -DXEUS_CPP_EMSCRIPTEN_WASM_BUILD=ON         \              
      -DCMAKE_FIND_ROOT_PATH=$PREFIX          \                  
      -DSYSROOT_PATH=$SYSROOT_PATH         \              
      -DCMAKE_CXX_FLAGS="-O0 -g -s -fsanitize=address -fno-omit-frame-pointer" ..    \
```

**Comment by kr-2003** [comment]
> Where would you be adding it ?
> 
> Here are some questions
> 
> 1. What is the pool of symbols that are being duplicated ? And from where (as in what is source_1 and source_2 and other sources from where these symbols are being duplicated)

Working on it 👍
